src/bin/cli.js

```diff
@@ -6,138 +6,147 @@ const chalk = require('chalk');
+ * Verifies the command line arguments, re-prompts in case of error. Parses AsyncApi File and returns the object representation.
  * @returns {Promise<*|null>}
  * @param rd
- * @param file
+ * @param asyncApiFilepath
  * @param scenarioFile
+ * @param basedir
  */
-const verifyInput_ParseAndLinkFiles =  async (rd, file,scenarioFile) => {
-  const handlingContext = this;
-  handlingContext.ready = false;
-  handlingContext.rd = rd;
-  handlingContext.file = file;
-  handlingContext.scenarioFile= scenarioFile;
-  const yamlJsonRegex = RegExp(/^.*\.(json|yaml)$/, 'gm');
-  handlingContext.yamlJsonRegex = yamlJsonRegex;
+const verifyInputGetData =  async (rd, asyncApiFilepath,scenarioFile,basedir) => {
+  const yamlJsonRegex = new RegExp(/^.*\.(json|yaml)$/, 'gm');
 
   console.log(chalk.blueBright(`
   Async api Fluffy-robot
   `));
   console.log('\nWelcome ');
 
-  if (!!file) {
-    if (!String(file).match(yamlJsonRegex)) {
-      console.log('\nPlease provide a correctly formatted filepath ex: ./myAsyncApi.json ./myAsyncApi.yaml:');
-      inputLoopAsyncApi(handlingContext);
-    } else {
-      handlingContext.file = file;
-      handlingContext.ready = true;
-      if (!scenarioFile) {
-        inputLoopScenario(handlingContext);
-      } else if (!String(scenarioFile).match(yamlJsonRegex)) {
-        inputLoopScenario(handlingContext);
-      } else {
-        handlingContext.scenarioFile = scenarioFile;
-        handlingContext.scenarioReady = true;
-        parseAsyncApi(handlingContext);
-      }
-    }
-  } else {
-    console.log('\nFilepath not provided');
-    inputLoopAsyncApi(handlingContext);
-  }
+  asyncApiFilepath = await inputLoopAsyncApi(rd,asyncApiFilepath,yamlJsonRegex,basedir);
 
-  return handlingContext.ready ? await handlingContext.ParsedAndFormated : null;
+  scenarioFile = await inputLoopScenario(rd,scenarioFile,yamlJsonRegex,basedir);
+
+  const structuredData = await  scenarioParserAndConnector(asyncApiFilepath,scenarioFile);
+
+  const availableServers = Object.keys(structuredData.servers);
+
+  structuredData.targetedServer = await inputLoopServer(availableServers);
+
+  return structuredData;
 };
 
 (async function Main ()  {
-  program.version('0.0.1', 'v', 'async-api performance tester cli version');
+  program.version('0.0.1', '-v', 'async-api performance tester cli version');
 
   program
-    .requiredOption('-f, --filepath <type>', 'The filepath of a async-api specification yaml or json file')
-    .requiredOption('-s, --scenario <type>', 'The filepath of a file defining a scenario based on the spec.')
-    .option('-b, --basedir <type>', 'The basepath from which relative paths are computed.\nDefaults to the directory where simulator.sh resides.');
+    .requiredOption('-f, --filepath <type>', 'The filepath of a async-api specification yaml or json asyncApiF')
+    .requiredOption('-s, --scenario <type>', 'The filepath of a asyncApiF defining a scenario based on the spec.')
```

Comment:
These two descriptions does not make sense to me 🤔 What is `asyncApiF`?

Reply:
ok